### PR TITLE
Add Spark list hashing Java tests

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR adds Java tests for the Spark list hashing feature added in #11292.

Depends on #11292.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java
Comment (ttnghia) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:785]: This causes memory leak: A `ColumnVector` construction should be wrapped in `try(....)`.

Comment (bdice) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:785]:
```suggestion
         ColumnVector expected = ColumnVector.spark32BitMurmurHash3(1868, new ColumnVector[]{strings, integers, doubles, floats, bools});
         ColumnVector result = ColumnVector.spark32BitMurmurHash3(1868, new ColumnView[]{structs})) {
```

Comment (bdice) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:785]: Good eye! I suggested the change below. I'll wait until #11292 is merged to try it.

**Review by ttnghia — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

**Comment by bdice** [comment]
@gpucibot merge

**Pull request merged**
